src/core/meshcop/leader.hpp

```diff
@@ -26,11 +26,131 @@
  *  POSSIBILITY OF SUCH DAMAGE.
  */
 
-#if OPENTHREAD_MTD
-#include "leader_mtd.hpp"
-#elif OPENTHREAD_FTD
-#include "leader_ftd.hpp"
-#else
-#error "Please define OPENTHREAD_MTD=1 or OPENTHREAD_FTD=1"
-#endif
+/**
+ * @file
+ *   This file includes definitions for a MeshCoP Leader.
+ */
+
+#ifndef MESHCOP_LEADER_HPP_
+#define MESHCOP_LEADER_HPP_
+
+#include <coap/coap_client.hpp>
+#include <coap/coap_server.hpp>
+#include <common/timer.hpp>
+#include <meshcop/tlvs.hpp>
+#include <net/udp6.hpp>
+#include <thread/mle.hpp>
+
+namespace Thread {
+namespace MeshCoP {
+
+OT_TOOL_PACKED_BEGIN
+class CommissioningData
+{
+public:
+    uint8_t GetLength(void) const {
```

Comment:
Understood - it technically better this way

If the policy is generally: Simple Getter/Setter is always inlined == And all other implementations should be done in an implementation file, then this does not follow that policy.

Developers are mostly"lemmings" and write via cut/paste techniques, they may/will question - why do we sometimes do XXX, and other times not do XXX - 

You know this, you determined this because you have deep knowledge about this and experimented - others do not have your depth here (ie: none of us experimented)

Going forward, I don't think we want to have this level of back and forth for every inlined C++ function.

Bottom line - I defer to others about the final call here.